DB/Item.lua

```diff
@@ -165,4 +165,83 @@ function Item:HasZone(entry)
 	return false
 end
 
+function Item:IsCollectionItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.COLLECTION
+end
+
+function Item:IsUsingCollectionProperties(entry)
+	return entry.collectedItemId ~= nil
+end
+
+function Item:IsArchaeologyItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ARCH
+end
+
+function Item:IsUsingArchaeologyProperties(entry)
+	return entry.raceId ~= nil
+end
+
+function Item:HasUseMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.USE
+end
+
+function Item:HasUseMethodProperties(entry)
+	return entry.items ~= nil
+end
+
+function Item:IsCovenantSpecificItem(entry)
+	return entry.requiresCovenant ~= nil
+end
+
+function Item:IsUsingCovenantProperties(entry)
+	return entry.requiredCovenantID ~= nil
+end
+
+function Item:IsZoneItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.ZONE
+end
+
+function Item:IsUsingZoneProperties(entry)
+	return entry.zones ~= nil
+end
+
+function Item:IsUsingFishingProperties(entry)
+	return entry.requiresPool ~= nil
+end
+
+function Item:IsFishingItem(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.FISHING
+end
+
+function Item:HasNPCMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.NPC
+end
+
+function Item:HasBossMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.BOSS
+end
+
+function Item:IsUsingNPCProperties(entry)
+	return entry.npcs ~= nil
+end
+
+function Item:HasSpecialMethod(entry)
+	return entry.method == CONSTANTS.DETECTION_METHODS.SPECIAL
+end
+
+function Item:IsUsingCoordsProperties(entry)
+	return entry.coords ~= nil
+end
+
+function Item:HasBothXYCoordinates(entry)
```

Comment:
Is this a descriptive name? If you intend to validate *all* the waypoints I'd use ``HasOnlyValidWaypoints()`` or something. If you want to see if it has *any* waypoints, maybe ``HasWaypoints()`` , or maybe you actually meant ``Has[AtLeastOne]ValidWaypoint()``? It's unwieldy, I guess.

Since you only check for x and y coordinates, it appears to be none of the above? Waypoints without a ``mapID`` are incomplete, as ``TomTom``'s API requires ``x``, ``y`` and ``mapID``, while Rarity itself can use the ``mapID`` without coordinates but not the other way around.

Reply:
Mostly rewritten in 4b8d97888ef3b226ee87f97c1d9797a2d5590871